hi-ml-histopathology/src/histopathology/utils/plots_utils.py

```diff
@@ -51,7 +53,18 @@ def save_confusion_matrix(results: ResultsType, class_names: Sequence[str], figu
     :param class_names: List of class names.
     :param figures_dir: The path to the directory where to save the confusion matrix.
     """
-    cf_matrix_n = confusion_matrix(results[ResultsKey.TRUE_LABEL], results[ResultsKey.PRED_LABEL], normalize="pred")
+    true_labels = [i.item() if isinstance(i, Tensor) else i for i in results[ResultsKey.TRUE_LABEL]]
+    pred_labels = [i.item() if isinstance(i, Tensor) else i for i in results[ResultsKey.PRED_LABEL]]
+    all_potential_labels = [i for i in range(len(class_names))]
+    present_labels_diff_expected = set(true_labels).union(set(pred_labels)).difference(set(all_potential_labels))
```

Comment:
I think it would be better to split this into two checks, one for true and one for predicted labels, so that the error message can be more specific?